agents/acu/acu_agent.py

```diff
@@ -918,15 +924,21 @@ def _run_specified_scan(self, session, times, azs, els, vas, ves, azflags, elfla
         self.log.info('No more points in the queue')
         current_az = self.data['broadcast']['Corrected_Azimuth']
         current_el = self.data['broadcast']['Corrected_Elevation']
-        while round(current_az - end_az, 1) != 0.:
-            self.log.info('Waiting to settle at azimuth position')
-            yield dsleep(0.1)
-            current_az = self.data['broadcast']['Corrected_Azimuth']
-        if not azonly:
-            while round(current_el - end_el, 1) != 0.:
-                self.log.info('Waiting to settle at elevation position')
+        az_attempt_number = 0
+        while az_attempt_number < 10:
+            while round(current_az - end_az, 1) != 0.:
+                self.log.info('Waiting to settle at azimuth position')
                 yield dsleep(0.1)
-                current_el = self.data['broadcast']['Corrected_Elevation']
+                az_attempt_number += 1
+                current_az = self.data['broadcast']['Corrected_Azimuth']
+        if not azonly:
+            el_attempt_number = 0
+            while el_attempt_number < 10:
+                while round(current_el - end_el, 1) != 0.:
+                    self.log.info('Waiting to settle at elevation position')
+                    yield dsleep(0.1)
+                    el_attempt_number += 1
+                    current_el = self.data['broadcast']['Corrected_Elevation']
```

Comment:
What is the idea behind these "attempt" loops?  They look like they might have some pathological behavior (specifically they will exit automatically after 1s even if the telescope is still not in position; but if the telescope _does_ reach target before 1s has elapsed then they will block forever).